.github/workflows/pr_infra.yaml

```diff
@@ -0,0 +1,63 @@
+name: pr_infra
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+    paths:
+      - 'infra/**'
+
+permissions: read-all
+
+jobs:
+  cloudformation:
+    runs-on: ubuntu-18.04 # for older version of awscli.
+    steps:
+      - id: file_changes
+        uses: trilom/file-changes-action@v1.2.4
```

Comment:
This one already doesn't run if you don't change something in infra/ .  (And I don't think it works at all if you do change something in infra/)